assets/section-main-product.css

```diff
@@ -1142,6 +1142,38 @@ a.product__text {
   border-color: rgb(var(--color-foreground));
 }
 
+.image--full-size {
+  cursor: zoom-out;
+  z-index: 1;
+  margin: 0;
+}
+
+.image--hover {
+  cursor: zoom-in;
+}
+
+.lightbox-button--none {
+  display: none;
+}
+
+.product-modal--hover, .product-modal--none {
+  display: block;
+}
+
+@media (hover: hover) {
+  .lightbox-button--hover {
+    display: none;
+  }
+
+  .product-modal--hover {
+    display: none;
+  }
+}
+
+.product__media-container {
```

Comment:
I see another styling declaration with this class on line `53` of the same file. I think we can probably remove it as there won't be any `slider-buttons` within this media container. 👍

Reply:
I agree that the other instance isn't being used for anything and we should remove it if we're keeping this class in this PR (if not it's not necessarily this pr's responsibility). However I'm wondering why we need this at all. 

Having `product__media` and `product__media-container` on the same element seems confusing. Could we not just target the existing classes instead? My other motivation for not adding this class is I'm working on a PR that's hoping to use a similar "product media container or wrapper" class in the thumbnail snippet.

Reply:
Initially, the `product__media` class doesn't have `overflow: hidden` on it. If I target `product__media` instead of creating a new class for these images alone, then we'd be hiding the overflow for other elements as well, and that might break the current design.

We're currently applying this class in our `noscript` code. If you think it's safe to force `overflow: hidden` on these too, then I can remove the `product__media-container` class. Alternatively, I can rename it to `product__media-image`, as it's only applied to images.

Reply:
So looking down at your changes on main-product.liquid. We're always applying the `product__media-container`, but we're _not_ including that in the noscript instances of the `product__media` class, right?

Could we target that particular instance in another way without a new class? `.product__modal-opener .product__media` or if my assumption above is correct, maybe even `.js .product__media`